src/controllers/users.js

```diff
@@ -0,0 +1,43 @@
+import { findByEmail, checkIfExistsInDb } from '../utils/searchDb';
+import { user, role } from '../models';
+
+/**
+ * @description Class based Controller for Roles
+ */
+class Users {
+/**
+ * @description Sets the permission for a given role to a particular resource
+ * @static
+ * @param {object} req object
+ * @param {object} res object
+ * @param {method} next method
+ * @returns {object } Sets Role for a given user
+ * @memberof Roles
+ * @type {object}
+ */
+  static async changeRole(req, res, next) {
+    const { email, roleId } = req.body;
+    try {
+      await findByEmail(email);
+      await checkIfExistsInDb(role, roleId, 'Role does not exist');
```

Comment:
checking if role exists is better done in a middleware. Your controller should only contain logic to actually update role in the db.

Reply:
Ok. Thanks for the insight